compiler-cli/src/dependencies.rs

```diff
@@ -434,23 +435,40 @@ async fn add_missing_packages<Telem: Telemetry>(
     let missing_packages = local.missing_local_packages(manifest, &project_name);
 
     let mut num_to_download = 0;
+
+    let missing_git_packages = missing_packages
+        .iter()
+        .copied()
+        .filter(|package| package.is_git())
+        .inspect(|_| {
+            num_to_download += 1;
+        })
+        .collect_vec();
+
     let mut missing_hex_packages = missing_packages
-        .into_iter()
+        .iter()
+        .copied()
         .filter(|package| package.is_hex())
         .inspect(|_| {
             num_to_download += 1;
         })
         .peekable();
 
     // If we need to download at-least one package
-    if missing_hex_packages.peek().is_some() {
+    if missing_hex_packages.peek().is_some() || !missing_git_packages.is_empty() {
```

Comment:
```suggestion
    if !missing_hex_packages.is_empty() || !missing_git_packages.is_empty() {
```

Let's make 'em match

Reply:
Thing is, `missing_git_packages` is a `Vec`, while `missing_hex_packages` is an `Iterator`. This is because the `Downloader::download_hex_packages` function takes an iterator and not a vector. To make them the same, we would need to collect the hex packages then iterate over them again, which seems slightly wasteful for a smal code change such as this.

Reply:
OH! I see. Thank you